zeus21/correlations.py

```diff
@@ -77,116 +90,228 @@ def get_xi_z0_lin(self):
-        self.Deltasq_xa = self.Deltasq_xa_lin + self._d_Pk_xa_nl * self._k3over2pi2 #note that it still has units of xa_avg
+        self._d_Pk_xa_nl_II = self.get_list_PS(self._II_deltaxi_xa, T21_coefficients.zintegral)
+        self._d_Pk_xa_nl_III = self.get_list_PS(self._III_deltaxi_xa, T21_coefficients.zintegral) #velocity correlations already embedded in nonlinear computation
+        self._d_Pk_xa_nl_IIxIII = self.get_list_PS(self._IIxIII_deltaxi_xa, T21_coefficients.zintegral)
+
+        self.Deltasq_xa_II = self.Deltasq_xa_lin_II + self._d_Pk_xa_nl_II * self._k3over2pi2 #note that it still has units of xa_avg
+        self.Deltasq_xa_III = self.Deltasq_xa_lin_III + self._d_Pk_xa_nl_III * self._k3over2pi2 #note that it still has units of xa_avg
+        self.Deltasq_xa_IIxIII = self.Deltasq_xa_lin_IIxIII + self._d_Pk_xa_nl_IIxIII * self._k3over2pi2 #note that it still has units of xa_avg
+
+
+        ##############################
+
+
+        #and same for xray
+        self._Pk_Tx_lin_II = self.windowxray_II**2 * Correlations._PklinCF
+        self._Pk_Tx_lin_III = self.windowxray_III**2 * Correlations._PklinCF ###HACFIX:+ self.windowxrayVel_III**2 * Correlations._PkEtaCF
+        self._Pk_Tx_lin_IIxIII = 2* self.windowxray_II * self.windowxray_III * Correlations._PklinCF #Pop IIxIII cross term doesn't have a velocity component
+
+        self.Deltasq_Tx_lin_II = self._Pk_Tx_lin_II * self._k3over2pi2
+        self.Deltasq_Tx_lin_III = self._Pk_Tx_lin_III * self._k3over2pi2
+        self.Deltasq_Tx_lin_IIxIII = self._Pk_Tx_lin_IIxIII * self._k3over2pi2
+
+        self._d_Pk_Tx_nl_II = self.get_list_PS(self._II_deltaxi_Tx, T21_coefficients.zintegral)
+        self._d_Pk_Tx_nl_III = self.get_list_PS(self._III_deltaxi_Tx, T21_coefficients.zintegral)
+        self._d_Pk_Tx_nl_IIxIII = self.get_list_PS(self._IIxIII_deltaxi_Tx, T21_coefficients.zintegral)
 
+        self.Deltasq_Tx_II = self.Deltasq_Tx_lin_II + self._d_Pk_Tx_nl_II * self._k3over2pi2
+        self.Deltasq_Tx_III = self.Deltasq_Tx_lin_III + self._d_Pk_Tx_nl_III * self._k3over2pi2
+        self.Deltasq_Tx_IIxIII = self.Deltasq_Tx_lin_IIxIII + self._d_Pk_Tx_nl_IIxIII * self._k3over2pi2
 
-        # #and same for xray
-        self._Pk_Tx_lin = self.windowxray**2 * Correlations._PklinCF
-        self.Deltasq_Tx_lin = self._Pk_Tx_lin * self._k3over2pi2
 
-        #self.deltaxi_Tx = self.get_Tx_deltaCF(Cosmo_Parameters, Correlations, T21_coefficients) #get them
-        self._d_Pk_Tx_nl = self.get_list_PS(self._deltaxi_Tx, T21_coefficients.zintegral)
-        self.Deltasq_Tx = self.Deltasq_Tx_lin + self._d_Pk_Tx_nl * self._k3over2pi2
+        ##############################
 
 
         #and their cross correlation
-        self._Pk_xaTx_lin = self.windowalpha * self.windowxray * Correlations._PklinCF
-        self.Deltasq_xaTx_lin = self._Pk_xaTx_lin * self._k3over2pi2
+        self._Pk_xaTx_lin_II = self.windowalpha_II * self.windowxray_II * Correlations._PklinCF
```

Comment:
I checked, this leads to a delay of 7e-5 seconds, so very negligible. Don't we want the user to get dimensionless Pop II/III/IIxIII power spectral components in case they want to debug?